core/src/main/java/org/infinispan/protostream/impl/TagWriterImpl.java

```diff
@@ -918,8 +957,206 @@ void writeBytes(ByteBuffer value) throws IOException {
       }
 
       @Override
-      void flush() throws IOException {
+      protected void flush() throws IOException {
          buffer.flushToStream(out);
       }
    }
+
+   private static class FixedVarintWrappedEncoder extends Encoder {
+      private final FixedVarintEncoder parentEncoder;
+      private final int originalPos;
+      private boolean closed;
+
+      private FixedVarintWrappedEncoder(FixedVarintEncoder parentEncoder) {
+         this.parentEncoder = parentEncoder;
+         this.originalPos = parentEncoder.skipFixedVarint();
+      }
+
+      @Override
+      protected void writeVarint32(int value) throws IOException {
+         parentEncoder.writeVarint32(value);
+      }
+
+      @Override
+      protected void writeVarint64(long value) throws IOException {
+         parentEncoder.writeVarint64(value);
+      }
+
+      @Override
+      protected void writeFixed32(int value) throws IOException {
+         parentEncoder.writeFixed32(value);
+      }
+
+      @Override
+      protected void writeFixed64(long value) throws IOException {
+         parentEncoder.writeFixed64(value);
+      }
+
+      @Override
+      protected void writeByte(byte value) throws IOException {
+         parentEncoder.writeByte(value);
+      }
+
+      @Override
+      protected void writeBytes(byte[] value, int offset, int length) throws IOException {
+         parentEncoder.writeBytes(value, offset, length);
+      }
+
+      @Override
+      protected void writeBytes(ByteBuffer value) throws IOException {
+         parentEncoder.writeBytes(value);
+      }
+
+      @Override
+      protected void close() throws IOException {
+         if (!closed) {
+            closed = true;
+            parentEncoder.writePositiveFixedVarint(originalPos);
+         }
+      }
+   }
+
+   private static class ArrayBasedWrappedEncoder extends Encoder {
```

Comment:
why doesn't it support `supportsFixedVarint`?

Reply:
I was lazy, adding :)